gitops/infrastructure/configs/gateway/tests/assert-attachment-boundary.sh

```
@@ -0,0 +1,16 @@
+set -euo pipefail
+
+runtime_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")/../runtime" && pwd)"
+gateway="$runtime_dir/gateway.yaml"
+
+kubectl kustomize "$runtime_dir" >/dev/null
+
+http_listener="$(awk '/- name: http$/,/    - name: https$/' "$gateway")"
+https_listener="$(awk '/- name: https$/,/^$/' "$gateway")"
+
+printf '%s\n' "$http_listener" | rg -q 'from: Same'
+! printf '%s\n' "$http_listener" | rg -q 'gateway-access:'
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)